packages/saber-plugin-amp/lib/index.js

```diff
@@ -5,25 +5,37 @@ const ID = 'amp'
 exports.name = ID
 
 exports.apply = api => {
+  let url = ''
+  let isAMP = false
+
+  api.hooks.beforeExportPage.tapPromise(ID, async () => {
+    isAMP = false
+  })
+
+  api.hooks.getDocumentData.tap(ID, data => {
+    if (isAMP) {
```

Comment:
this doesn't really work becuase `afterExportPage` is called after `getDocumentData`.

you can access `context` directly in this hook now https://github.com/saberland/saber/commit/0f6669eb0b2714c971b62a039dbb0d5296cb858b

Reply:
In [vue-renderer](https://github.com/saberland/saber/blob/master/packages/saber/vue-renderer/lib/index.js) that's true but in this plugin `getDocumentData` is called after `afterExportPage` event. check line 30 down. but it's great to access `context` in the `getDocumentData` hook I wont need to use the `url` variable anymore

Reply:
the `tap` order doesn't affect the `call` order 😅

Reply:
strange it's generating amp pages without problem, whatever I will get ride of `api.hooks.afterExportPage.tapPromise` and use just `api.hooks.getDocumentData.tap`.

Reply:
@egoist I just understood what you mean (maybe.....)  I used `isAMP` variable to make sure that the changes that I make to the `data` happen only after a normal page is exported then I change it to `isAMP = context.amp` inside `api.hooks.afterExportPage.tapPromise` so when `api.getDocumentData(context)` is executed in line 36 `api.hooks.getDocumentData.tap` get triggered again

Reply:
@egoist if you want me to move the code inside `api.hooks.afterExportPage.tapPromise` to `api.hooks.getDocumentData.tap` tell me so but I will have to change some small stuff in vue-renderer (I will have to pass `exportedPage` to `getDocumentData` which will make the core code a little bit messy)

Reply:
@egoist check the last commit I added some explanation